modules/bmo-about-the-baremetalhost-resource.adoc

```diff
@@ -0,0 +1,270 @@
+// This is included in the following assemblies:
+//
+// post_installation_configuration/bare-metal-configuration.adoc
+
+:_content-type: REFERENCE
+[id="about-the-baremetalhost-resource_{context}"]
+= About the BareMetalHost resource
+
+Metal^3^ introduces the concept of the `BareMetalHost` resource, which defines a physical host and its properties. The `BareMetalHost` resource contains two sections:
+
+. The bare metal host specification.
+. The bare metal host's current status.
+
+[discrete]
+== The BareMetalHost specification
+
+The `spec` section of the `BareMetalHost` resource defines the desired state of the host.
+
+.BareMetalHost spec
```

Comment:
For the baremetalHost I think the most definitive way to get the fields for 4.10 (as some have been added) is to use the api definition in code here https://github.com/openshift/baremetal-operator/blob/release-4.10/apis/metal3.io/v1alpha1/baremetalhost_types.go#L316 (for spec) and https://github.com/openshift/baremetal-operator/blob/release-4.10/apis/metal3.io/v1alpha1/baremetalhost_types.go#L700 (for status). The comments can be used their for description or the description can be taken from https://github.com/openshift/baremetal-operator/blob/master/docs/api.md#baremetalhost. but that doesn't have the complete set of fields.

I think the fields should also be listed alphabetically as that is how k8s shows them when you do "oc describe".

Reply:
Ok. I've sorted them, and added firmware, raid and errorType fields.